Use url-connection-client for dynamodb and graal

timyates opened the pull request:
The apache client doesn't work with native compilation, so if the user selects
dynamodb and graalvm then we switch to the url-connection one, and exclude the other
from the dependencies

This also fixes exclusions for Gradle which were not previously required

Pushed for 3.5.x incase we do a release of that as mentioned in the team meeting

timyates committed:
Use url-connection-client for dynamodb and graal

The apache client doesn't work with native compilation, so if the user selects
dynamodb and graalvm then we switch to the url-connection one, and exclude the other
from the dependencies

This also fixes exclusions for Gradle which were not previously required

sdelamo commented on starter-core/src/main/java/io/micronaut/starter/build/gradle/GradleDependency.java:
```diff
@@ -102,6 +102,18 @@ public String toSnippet() {
         if (isPom() && isKotlinDSL) {
             snippet += ")";
         }
+        if (getExclusions() != null && !getExclusions().isEmpty()) {
```
We have a Micronaut `CollectionsUtils::isEmpty` to simplify this check

sdelamo commented on starter-core/src/main/java/io/micronaut/starter/feature/aws/DynamoDb.java:
```diff
@@ -35,11 +35,29 @@
 public class DynamoDb implements AwsFeature {
     public static final String ARTIFACTID_DYNAMODB = "dynamodb";
+
+    private static final Dependency.Builder URL_CONNECTION_CLIENT = Dependency.builder()
```
move this statics to AwsSdkV2 feature

sdelamo reviewed (commented):
I created a PR

https://github.com/micronaut-projects/micronaut-starter/pull/1320

timyates committed:
Address feedback

timyates committed:
Use isNotEmpty

sdelamo committed:
Move logic to utils (#1320)

* refactor: move logic to utils

This moves the logic to a Utils class so that we are able to apply this to other AWS SDK related features not just to dynamodb. E.g. S3.

* use CollectionsUtils

* Fix my bad merge

Co-authored-by: Tim Yates <tim.yates@gmail.com>
